__tests__/app.test.js

```diff
@@ -323,3 +324,50 @@ describe("GET /api/articles/:article_id/comments", () => {
       });
   });
 });
+
+describe("POST /api/articles/:article_id/comments", () => {
+  test("should return the posted comment", () => {
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send({ username: "butter_bridge", body: "sample text" })
+      .expect(201)
+      .then((response) => {
+        expect(response.body).toEqual(
+          expect.objectContaining({
+            author: "butter_bridge",
+            body: "sample text",
+            article_id: 1,
+            created_at: expect.any(String),
+            votes: 0,
+          })
+        );
+      });
+  });
+  test("should return 400 Bad Request for missing required fields in input", () => {
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send({})
+      .expect(400)
+      .then((response) => {
+        expect(response._body.msg).toBe("Bad request");
+      });
+  });
+  test("should return 400 Bad Request for incorrect types in input", () => {
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send({ username: 1, body: 1 })
+      .expect(400)
+      .then((response) => {
+        expect(response._body.msg).toBe("Bad request");
+      });
+  });
+  test("should return 400 Bad Request for one missing input field", () => {
```

Comment:
We also need to deal with the usual: what if the request is to an article that is the right type but doesn't exist in the database and what if it's passed in the wrong type?